File: src/features/typing/data/ts/min_stack.ts
class MinStack {
    private stack: number[] = [];
    private minStack: number[] = [];

    push(val: number): void {
        this.stack.push(val);
        let min = val;
        if (this.minStack.length > 0 && this.minStack[this.minStack.length - 1] < min) {
            min = this.minStack[this.minStack.length - 1];
        }
        this.minStack.push(min);
    }

    pop(): void {
        this.stack.pop();
        this.minStack.pop();
    }

    top(): number {
        return this.stack[this.stack.length - 1];
    }

    getMin(): number {
        return this.minStack[this.minStack.length - 1];
    }
}

File: src/features/typing/data/go/min_stack.go
package main

type MinStack struct {
	stack    []int
	minStack []int
}

func MinStackConstructor() MinStack {
	return MinStack{}
}

func (s *MinStack) Push(val int) {
	s.stack = append(s.stack, val)
	min := val
	if len(s.minStack) > 0 && s.minStack[len(s.minStack)-1] < min {
		min = s.minStack[len(s.minStack)-1]
	}
	s.minStack = append(s.minStack, min)
}

func (s *MinStack) Pop() {
	s.stack = s.stack[:len(s.stack)-1]
	s.minStack = s.minStack[:len(s.minStack)-1]
}

func (s *MinStack) Top() int {
	return s.stack[len(s.stack)-1]
}

func (s *MinStack) GetMin() int {
	return s.minStack[len(s.minStack)-1]
}
